src/app/api/github.ts

```diff
@@ -0,0 +1,65 @@
+import { HTTPClient } from "./httpClient";
+import { GitHubRepository } from "../models/repository";
+
+export class GitHubAPI {
+    httpClient = new HTTPClient("api.github.com")
+
+    public tags(owner: string, repo: string): Promise<[string]> {
+        return this.tagURL(owner, repo).then(url => {
+            const items = url.split(this.httpClient.host)
+            const repoPath = items[items.length-1]
+            return this.repositorySourceInfo(repoPath)
+        });
+    }
+
+    public branches(owner: string, repo: string): Promise<[string]> {
+        return this.branchURL(owner, repo).then(url => {
+            const items = url.split(this.httpClient.host)
+            const repoPath = items[items.length-1]
+            return this.repositorySourceInfo(repoPath)
+        });
+    }
```

Comment:
I don't understand what you are doing here. I think both functions can be reduced to:

```typescript
this.repositorySourceInfo(`/repos/${owner}/${repo}/branches`)
this.repositorySourceInfo(`/repos/${owner}/${repo}/tags`)
```

and you can get rid of several other functions below.

Reply:
I will review the b. logic I was using the full-rest Github response, but I agree it can be simplified using directly the endpoint.